pipes/WeCount-ODC.js

```diff
@@ -0,0 +1,121 @@
+ * https://data.ontario.ca/dataset/covid-19-assessment-centre-locations
+ * @param {String} dataSourceURL - The URL to the webpage where the information of the data file is published
+ * @return {Object} An object with members
+ *     {String} downloadURL - The url to download the new data file
+ *     {String} date - The last updated date of the data file as rendered in the manifest page which is in ISO 8601 date format
+ */
+fluid.dataMonitor.scrapeRepositoryPage = async function (dataSourceURL) {
+
+    let res = await axios.get(dataSourceURL);
+    let data = res.data;
+    let dom = new JSDOM(data);
+
+    const findElements = function (selector) {
+        return dom.window.document.querySelectorAll(selector);
+    };
+
+    let downloadLink;
+    let as = findElements("a.dataset-download-link");
+    for (let a of as) {
+        let link = a.getAttribute("href");
+        if (link.slice(-4) === ".csv") {
+            downloadLink = link;
+            break;
+        }
+    }
+
+    // find the last date the dataset was updated
+    let lastUpdate;
+    let tableHeaders = findElements("th.dataset-label");
+    for (let header of tableHeaders) {
+        if (header.innerHTML === "Last Validated Date") {
+            lastUpdate = header.parentElement.querySelector("td.dataset-details").innerHTML.trim();
+            break;
+        }
+    }
+
+    return {
+        downloadURL: downloadLink,
+        date: lastUpdate
+    };
+};
+
+// Returns the bare filename without extension
+fluid.dataMonitor.trunkName = function (path) {
+    var dotpos = path.lastIndexOf(".");
+    var slashpos = path.lastIndexOf("/");
+    return path.substring(slashpos + 1, dotpos);
+};
+
+
+function generateMergedFileName(ODCDate, assessmentFile) {
+    var trunkName = fluid.dataMonitor.trunkName(assessmentFile);
+    var assessDate = trunkName.slice(-10);
+    return "merged_ODC_" + ODCDate.replace(/-/g, "_") + "_WeCount_" + assessDate + ".csv";
+};
+
+fluid.defaults("fluid.dataMonitor.getMergedFilename", {
+    gradeNames: "fluid.dataPipe"
+});
+
+/** Compute the filename to be used to represent a merged result of the WeCount and ODC datasets.
+ * @param {Object} options - Accepts a structure
+ *     ODCCoordinates: "{ODCCoordinates}.data"
+ *     WeCountFilePath: "{WeCount}.options.filePath"
+       folder: "{gitConfig}.mergedFolder
+ * @return {Object} A structure holding the required path in member `filePath`.
+ */
+fluid.dataMonitor.getMergedFilename = function (options) {
+    return {
+        filePath: options.folder + generateMergedFileName(options.ODCCoordinates.date, options.WeCountFilePath)
+    };
+};
+
+/** Given a pending set of file writes, append a small index file named `latest.json` determining which is the
+ * latest dated file just written.
+ * @param {FileEntry} fileOptions - Overall file options for the file being written
+ * @return {FileEntry} An additional `FileEntry` element recording the additional index file to be written
```

Comment:
The returned structure should no longer be `{FileEntry}`, which is the structure of the input argument.

Reply:
Actually the two structures do have the same type